File: python/073.py
#!/usr/bin/env python3
LIMIT_D = 12_000
count = 0

for d in range(1, LIMIT_D + 1):
    n_min = d // 3 + 1
    n_max = (d - 1) // 2

    if n_min <= n_max:
        for n in range(n_min, n_max + 1):
            if __import__('math').gcd(n, d) == 1:
                count += 1

print(count)

File: c/073.c
/* Project Euler 073 - Counting fractions in a range */
#include <stdio.h>

static int gcd(int a, int b) {
    while (b) {
        int t = b;
        b = a % b;
        a = t;
    }
    return a;
}

int main(void) {
    int limit_d = 12000;
    int count = 0;

    for (int d = 1; d <= limit_d; d++) {
        int n_min = d / 3 + 1;
        int n_max = (d - 1) / 2;

        for (int n = n_min; n <= n_max; n++) {
            if (gcd(n, d) == 1) {
                count++;
            }
        }
    }

    printf("%d\n", count);
    return 0;
}
